openbas-api/src/main/java/io/openbas/scheduler/jobs/InjectsExecutionJob.java

```diff
@@ -167,62 +167,75 @@ private void executeInject(ExecutableInject executableInject) {
                         status.setCommandsLines(atomicTestingService.getCommandsLinesFromInject(inject));
                         injectStatusRepository.save(status);
                     }
-                    return;
+                    newExecutableInject = this.executionExecutorService.launchExecutorContext(executableInject, inject);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
                 }
+            }
+            if (externalInjector.isExternal()) {
+                executeExternal(newExecutableInject);
+            } else {
+                executeInternal(newExecutableInject);
+            }
+        }, () -> setInjectStatusWhenNoInjectorContractExists(inject));
+    }
 
-                Injector externalInjector = injectorRepository.findByType(injectorContract.getInjector().getType()).orElseThrow();
-                LOGGER.log(Level.INFO, "Executing inject " + inject.getInject().getTitle());
-                // Executor logics
-                ExecutableInject newExecutableInject = executableInject;
-                if (Boolean.TRUE.equals(injectorContract.getNeedsExecutor())) {
-                    try {
-                        // Status
-                        if (inject.getStatus().isEmpty()) {
-                            InjectStatus status = new InjectStatus();
-                            status.setName(ExecutionStatus.EXECUTING);
-                            status.setTrackingSentDate(Instant.now());
-                            status.setInject(inject);
-                            status.setCommandsLines(atomicTestingService.getCommandsLinesFromInject(inject));
-                            injectStatusRepository.save(status);
-                        } else {
-                            InjectStatus status = inject.getStatus().get();
-                            status.setName(ExecutionStatus.EXECUTING);
-                            status.setTrackingSentDate(Instant.now());
-                            status.setCommandsLines(atomicTestingService.getCommandsLinesFromInject(inject));
-                            injectStatusRepository.save(status);
-                        }
-                        newExecutableInject = this.executionExecutorService.launchExecutorContext(executableInject, inject);
-                    } catch (InterruptedException e) {
-                        throw new RuntimeException(e);
-                    }
-                }
-                if (externalInjector.isExternal()) {
-                    executeExternal(newExecutableInject);
-                } else {
-                    executeInternal(newExecutableInject);
-                }
-            });
-        }
+    private void setInjectStatusWhenNoInjectorContractExists(Inject inject) {
+        InjectStatus status = new InjectStatus();
+        status.getTraces().add(InjectStatusExecution.traceError("The inject has no injector contract"));
```

Comment:
Could you verify if inject doesnt have another status?

Reply:
Done